src/query/sql/src/evaluator/block_operator.rs

```diff
@@ -151,98 +152,169 @@ impl BlockOperator {
+                                                        .build();
+                                                    }
+                                                    Column::Nullable(box nullable_column) => {
+                                                        let mut column_builder =
+                                                            NullableColumnBuilder::from_column(
+                                                                (*nullable_column).clone(),
+                                                            );
+                                                        (0..(max_nums_per_row[i] - repeat_times))
+                                                            .for_each(|_| {
+                                                                column_builder.push_null();
+                                                            });
+                                                        *field = Column::Nullable(Box::new(
+                                                            column_builder.build(),
+                                                        ));
+                                                    }
+                                                    _ => unreachable!(),
+                                                }
+                                            }
                                         }
-                                        _ => unreachable!(),
+                                    } else {
+                                        row_result = Value::Column(
+                                            ColumnBuilder::repeat(
+                                                &ScalarRef::Tuple(vec![ScalarRef::Null]),
+                                                max_nums_per_row[i],
+                                                srf_expr.data_type(),
+                                            )
+                                            .build(),
+                                        );
                                     }
+
+                                    let block_entry =
+                                        BlockEntry::new(srf_expr.data_type().clone(), row_result);
+                                    result_data_blocks.push(DataBlock::new(
+                                        vec![block_entry],
+                                        max_nums_per_row[i],
+                                    ))
+                                }
+                                let data_block = DataBlock::concat(&result_data_blocks)?;
+                                debug_assert!(data_block.num_rows() == total_num_rows);
+                                let block_entry = BlockEntry::new(
+                                    data_block.get_by_offset(0).data_type.clone(),
+                                    data_block.get_by_offset(0).value.clone(),
+                                );
+                                if block_is_empty {
```

Comment:
There are several places where `block_is_empty` is checked, is it possible to put the `block_entry` into a vector first and then construct the `block` at the end?

Reply:
Due to `unused_indices` , we can't directly get the first `BlockEntry`.